e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/ConstraintsTests.java

```diff
@@ -258,7 +258,8 @@ void runConstraintsOnOldSimulation() throws IOException {
 
     // Expect no violations on the new simulation
     final var newConstraintResults = hasura.checkConstraints(planId, newSimDatasetId);
-    assertEquals(0, newConstraintResults.size());
+    assertEquals(1, newConstraintResults.size());
+    assertEquals(null, newConstraintResults.get(0).result());
```

Comment:
Why does `checkConstaints` now return `NULL` on success?

Reply:
The data we get back from the server is below:

```js
{
status : success,
results: {},
errors: []
}
```

Converting this to java

```java
ConstraintRecord[success=true, result=null, errors=[]]
```

I made the conversion between an empty object from JS to java as `null`. 

I can change this to optional unless you have another suggestion?

Reply:
It should be a proper `ConstraintResult` object where the `violations` field is empty (as, for example, there still could've been gaps)

Reply:
Done, Return a ConstraintResult object